posts-router.js

```diff
@@ -0,0 +1,146 @@
+const express = require('express');
+
+const db = require('./data/db');
+
+const router = express.Router();
+
+router.get('/', (req, res) => {
+  db
+    .find()
+    .then(posts => {
+      res
+        .status(200)
+        .json({
+          success: true, posts
+        })
+    })
+    .catch(error => {
+      res
+        .status(500)
+        .json({
+          success: false,
+          error: 'The posts information could not be retrieved.'
+        })
+    });
+})
+
+router.get('/:id', (req, res) => {
+  db
```

Comment:
try throwing your search for the post - the else error above your initial request. it will save the processing time.